### PR TITLE
[ACR] Fixes and improvements after Knack conversion

### PR DESCRIPTION
**Pull request by djyou**
Fixes https://github.com/Azure/azure-cli/issues/5229

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by djyou** [comment]
@tjprescott This is a more comprehensive PR than the previous one (closed) with fixes and improvements. The added tests can help prevent regression in the future. Can you please review and merge it to unblock users? Thanks.

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_docker_utils.py [hunk 1]
```diff
@@ -130,7 +130,7 @@ def _get_credentials(cli_ctx,
     :param str repository: Repository for which the access token is requested
     :param str permission: The requested permission on the repository, '*' or 'pull'
     """
-    registry, _ = get_registry_by_name(cli_ctx, registry_name, resource_group_name)
+    registry, resource_group_name = get_registry_by_name(cli_ctx, registry_name, resource_group_name)
```
Comment (tjprescott) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_docker_utils.py]: Does the resource group name get transformed somehow?  Why do you need it to return from the method when you are supplying it as a parameter?

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/tests/test_acr_commands_mock.py [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+            urlencode({
+                'grant_type': 'access_token_refresh_token',
+                'service': 'testregistry.azurecr.io',
+                'tenant': 'testtenant',
+                'access_token': 'aadaccesstoken',
+                'refresh_token': 'aadrefreshtoken'
+            }),
+            headers={'Content-Type': 'application/x-www-form-urlencoded'})
+
+        get_access_credentials(cmd.cli_ctx, 'testregistry', None, None, None, 'testrepository')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/tests/test_acr_commands_mock.py]: Just to verify, does this eventually call the _get_credentials method where the login regression occurred?

Comment (djyou) [reply src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/tests/test_acr_commands_mock.py]: Yes. Both `get_access_credentials` and `get_login_credentials` call `_get_credentials`.

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_docker_utils.py (hunk 1 above)
Comment (djyou) [reply src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_docker_utils.py]: The incoming `resource_group_name` can be empty but the returned must have a value. The method `get_registry_by_name` has a different logic depending on whether `resource_group_name` is `None`.

**Comment by tjprescott** [comment]
Thanks @djyou !!

**Pull request merged**